### PR TITLE
several fixes

### LINKED ISSUE #289: Error deploying secure registry
## Issue 

When we execute the following command on macos, then we got the error `sudo: service: command not found`
The script also requests to provide the admin password able to execute on a mac: `sudo vi /etc/hosts`

```
./kind.sh install --secure-registry
...
==== Restarting containerd
Copy the client.crt to the docker cert.d folder
Password: 
sudo: service: command not found
```

### LINKED ISSUE #291: unknown shorthand flag: 'f' in -f when removing docker network on macos
```
unknown shorthand flag: 'f' in -f
See 'docker network rm --help'.
```

### LINKED ISSUE #297: Add option to skip ingress installation when deploying kind
Add a parameter like `--skip-ingress-installation`.

### PR DESCRIPTION
**Pull request by jacobdotcosta**
Closes #291 
Closes #289 
Closes #297 

Other changes and fixes:

* More verbose result message
* Check that either `docker` or `podman` are installed
* When using `docker` in `linux`, check if the docker service is started

### CODE REVIEW

File: kind/kind.sh [hunk 1]
```diff
@@ -414,59 +471,85 @@ EOF
...
         done <<< "$containers"
 
         log_message "1" "Copy the client.crt to the docker cert.d folder"
-        sudo mkdir -p /etc/docker/certs.d/${SERVER_IP}.sslip.io:5000
-        sudo cp $certfile /etc/docker/certs.d/${SERVER_IP}.sslip.io:5000/ca.crt
-        sudo service docker restart
+        if [[ "$OSTYPE" != "darwin"* ]]; then
+            sudo mkdir -p /etc/docker/certs.d/${SERVER_IP}.sslip.io:${REGISTRY_PORT}
+            sudo cp $certfile /etc/docker/certs.d/${SERVER_IP}.sslip.io:${REGISTRY_PORT}/client.crt
+        else
```
Comment (cmoulliard) [inline kind/kind.sh:494]: This step can be removed as the else block works on macos or linux ;-)

**Review by cmoulliard — COMMENTED** [review]
Can you also do the following change ?

- Use UPPERCASE and not lowercase for this parameter `registry_name`
- Remove `registry_server` as we dont use it

File: kind/kind.sh (hunk 1 above)
Comment (jacobdotcosta) [reply kind/kind.sh]: fixed

File: kind/kind.sh [hunk 2]
```diff
@@ -488,41 +488,35 @@ EOF
...
-            mkdir -p $HOME/.docker/certs.d/<${SERVER_IP}.sslip.io:${REGISTRY_PORT}
-            cp $certfile $HOME/.docker/certs.d/<${SERVER_IP}.sslip.io:${REGISTRY_PORT}/client.crt
-            cp $HOME/.registry/certs/${registry_name}/client.key $HOME/.docker/certs.d/<${SERVER_IP}.sslip.io:${REGISTRY_PORT}/client.key
-            #sudo service docker restart
-            eval ${DOCKER_RESTART_COMMAND}
-        fi
+        mkdir -p $HOME/.docker/certs.d/<${SERVER_IP}.sslip.io:${REGISTRY_PORT}
+        cp $certfile $HOME/.docker/certs.d/<${SERVER_IP}.sslip.io:${REGISTRY_PORT}/client.crt
+        cp $HOME/.registry/certs/${REGISTRY_NAME}/client.key $HOME/.docker/certs.d/<${SERVER_IP}.sslip.io:${REGISTRY_PORT}/client.key
+        eval ${DOCKER_RESTART_COMMAND}
```
[the remark this answers is not in the document]
Comment (jacobdotcosta) [reply kind/kind.sh:494]: As of the [docker insecure registry documentation](https://docs.docker.com/registry/insecure/#deploy-a-plain-http-registry) it is required to restart docker, at least in Linux.

File: kind/kind.sh [hunk 3]
```diff
@@ -488,41 +488,35 @@ EOF
...
-        else
-            mkdir -p $HOME/.docker/certs.d/<${SERVER_IP}.sslip.io:${REGISTRY_PORT}
-            cp $certfile $HOME/.docker/certs.d/<${SERVER_IP}.sslip.io:${REGISTRY_PORT}/client.crt
-            cp $HOME/.registry/certs/${registry_name}/client.key $HOME/.docker/certs.d/<${SERVER_IP}.sslip.io:${REGISTRY_PORT}/client.key
-            #sudo service docker restart
-            eval ${DOCKER_RESTART_COMMAND}
-        fi
+        mkdir -p $HOME/.docker/certs.d/<${SERVER_IP}.sslip.io:${REGISTRY_PORT}
+        cp $certfile $HOME/.docker/certs.d/<${SERVER_IP}.sslip.io:${REGISTRY_PORT}/client.crt
+        cp $HOME/.registry/certs/${REGISTRY_NAME}/client.key $HOME/.docker/certs.d/<${SERVER_IP}.sslip.io:${REGISTRY_PORT}/client.key
```
[the remark this answers is not in the document]
Comment (jacobdotcosta) [reply kind/kind.sh]: @cmoulliard Shouldn't all references to `${SERVER_IP}.sslip.io` be replaced with `${REGISTRY_NAME}.sslip.io`?

[the remark this answers is not in the document]
Comment (jacobdotcosta) [reply kind/kind.sh]: fixed

File: kind/kind.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (jacobdotcosta) [reply kind/kind.sh:494]: I've cleared the `DOCKER_RESTART_COMMAND` for darwin so the process isn't blocked.

**Review by cmoulliard — APPROVED** [review]

**Pull request merged**